tests/playbooks/tests_eth_pci_address_match.yml

```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- hosts: all
+
+- name: Test configuring ethernet devices
+  hosts: all
+  vars:
+    type: veth
+    interface: eth0
+
+  tasks:
+    - name: "set type={{ type }} and interface={{ interface }}"
+      set_fact:
+        type: "{{ type }}"
+        interface: "{{ interface }}"
+    - include_tasks: tasks/show_interfaces.yml
+    - include_tasks: tasks/manage_test_interface.yml
+      vars:
+        state: present
+    - include_tasks: tasks/assert_device_present.yml
+
+    - import_role:
+        name: linux-system-roles.network
+      vars:
+        network_connections:
+          - name: "{{ interface }}"
+            # interface_name: eth0
+            persistent_state: present
+            type: ethernet
+            interface_path:
+              - pci-0000:00:03.0
```

Comment:
I assume you use `pci-0000:00:03.0` address which does not actually exists in CI environment and this test case only
check whether NM created the desired connection.

Two concerns:
 * Please use something like `0001:00:00.0` and a comment indicate this is psude PCI address only for test purpose.
 * Please remove the requirement of veth eth0 above and use `psudo-pci-test-only` to make sure NM does not use interface name as matching.

Reply:
Updated, thanks.